FGEGraphics/UISystem/UINumberInputLabel.cs

```diff
@@ -38,16 +27,57 @@ public class UINumberInputLabel(double initial, bool integer, string format, UIE
     public static readonly AsciiMatcher DecimalMatcher = new(AsciiMatcher.Digits + "-.e");
 
     /// <summary>Whether the label should be an integer instead of a decimal.</summary>
-    public bool Integer = integer;
+    public bool Integer;
 
     /// <summary>The format string to apply to the label on submission.</summary>
-    public string Format = format;
+    public string Format;
 
     /// <summary>The character matcher for this number label type.</summary>
     public AsciiMatcher CharacterMatcher => Integer ? IntegerMatcher : DecimalMatcher;
 
-    /// <summary>The decimal value of the label.</summary>
-    public double Value => double.Parse(TextContent);
+    /// <summary>Data internal to a <see cref="UINumberInputLabel"/> instance.</summary>
+    public new struct InternalData
```

Comment:
should ideally just give the internal structs distinct names rather than overlapping them with `new`